front/src/components/LoginPanel/LoginPanel.test.js

```diff
@@ -0,0 +1,36 @@
+import { render, screen } from "@testing-library/react";
+import { within } from "@testing-library/dom";
+import userEvent from "@testing-library/user-event";
+import { MockedProvider } from "@apollo/client/testing";
+import LoginPanel from "./LoginPanel";
+
+describe("LoginPanel", () => {
+    var container;
+    beforeEach(() => {
+        container = render(
+            <MockedProvider mocks={[]} addTypename={false}>
+                <LoginPanel />
+            </MockedProvider>
+        ).container;
+    });
+
+    it("should render login panel by default", () => {
+        let loginForm = screen.getByTestId("loginForm");
+        expect(loginForm).toBeInTheDocument();
+    });
+
+    it("should have two buttons to change subpanels", () => {
+        let buttonsDiv = container.querySelector(".buttons");
```

Comment:
Querying by div or other properties invisible for user is the least preferable option, but in this case is necessary to distinguish from buttons rendered by nested components.
Consider changing class name to something specific to this and only this div or even adding some other easily-distinguishable property (role, id, etd).